rustls/src/server/server_conn.rs

```diff
@@ -758,6 +769,31 @@ impl Acceptor {
     }
 }
 
+/// Represents a TLS alert resulting from handling the client's `ClientHello` message.
+///
+/// When [`Acceptor::accept()`] returns an error, it yields an `AcceptedAlert` such that the
+/// application can communicate failure to the client via [`AcceptedAlert::write()`].
+pub struct AcceptedAlert(ChunkVecBuffer);
+
+impl AcceptedAlert {
+    /// Send the alert to the client.
+    pub fn write(&mut self, wr: &mut dyn io::Write) -> Result<usize, io::Error> {
+        self.0.write_to(wr)
+    }
+}
+
+impl From<ConnectionCommon<ServerConnectionData>> for AcceptedAlert {
+    fn from(conn: ConnectionCommon<ServerConnectionData>) -> Self {
+        Self(conn.core.common_state.sendable_tls)
+    }
+}
+
+impl Debug for AcceptedAlert {
```

Comment:
I think we might want `Debug` on `Accepted` as well. I was surprised that writing `let err_result = acceptor.accept().unwrap_err();` apparently requires it:

```
error[E0277]: `Accepted` doesn't implement `std::fmt::Debug`
    --> rustls/tests/api.rs:5408:40
     |
5408 |     let err_result = acceptor.accept().unwrap_err();
     |                                        ^^^^^^^^^^ `Accepted` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
     |
```